src/embedded/embeddedManager.ts

```diff
@@ -114,9 +111,9 @@ export class EmbeddedManager {
       this.notifyUpdateDelegates();
     }
     for (let i = 0; i < msgsList.length; i++) {
-      const messages = {} as IEmbeddedMessageData;
-      messages.messageId = msgsList[i].metadata.messageId;
-      await trackEmbeddedMessageReceived(messages);
+      const message = {} as EmbeddedMessage;
+      message.messageId = msgsList[i].metadata.messageId;
+      await trackEmbeddedReceived(message.messageId);
```

Comment:
Also, it doesn't look like we're using message anywhere further (though I'm not in an IDE so can't say for certain), and the typecasting doesn't look like it's doing anything (it's not like message.messageId is a special type, just a string), so perhaps could be simplified to:
```suggestion
      const { messageId } = msgsList[i].metadata;
      await trackEmbeddedReceived(messageId);
```